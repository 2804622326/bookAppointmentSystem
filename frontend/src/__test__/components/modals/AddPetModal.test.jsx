import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import AddPetModal from '../../../components/modals/AddPetModal.jsx';

test('adds pet with filled information', () => {
  const onAddPet = jest.fn();
  const onHide = jest.fn();
  const { asFragment } = render(
    <AddPetModal show onHide={onHide} onAddPet={onAddPet} appointmentId={1} />
  );
  fireEvent.change(screen.getByLabelText(/name/i), { target: { value: 'Tom' } });
  fireEvent.change(screen.getByLabelText(/age/i), { target: { value: '3' } });
  fireEvent.click(screen.getByText(/add pet/i));
  expect(onAddPet).toHaveBeenCalled();
  expect(asFragment()).toMatchSnapshot();
});
