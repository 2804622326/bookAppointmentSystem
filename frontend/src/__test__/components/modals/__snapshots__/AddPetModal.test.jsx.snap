// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds pet with filled information 1`] = `<DocumentFragment />`;
